pkg/controller/command/verifiable/command.go

```diff
@@ -245,3 +250,46 @@ func (o *Command) GetCredentials(rw io.Writer, req io.Reader) command.Error {
 
 	return nil
 }
+
+// GeneratePresentation generates verifiable presentation from a verifiable credential.
+func (o *Command) GeneratePresentation(rw io.Writer, req io.Reader) command.Error {
+	request := &Credential{}
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, generatePresentationCommandMethod, "request decode : "+err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	// TODO https://github.com/hyperledger/aries-framework-go/issues/1316 Add keys for proof
+	//  verification as options to the function.
+	vc, _, err := verifiable.NewCredential([]byte(request.VC))
```

Comment:
How do you know which type of proof to create?

Reply:
cc: @kdimak

Reply:
Created an issue https://github.com/hyperledger/aries-framework-go/issues/1422